simapp/app_test.go

```diff
@@ -0,0 +1,53 @@
+package simapp
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/tendermint/tendermint/libs/log"
+	dbm "github.com/tendermint/tm-db"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+func TestSimAppExport(t *testing.T) {
+	db := dbm.NewMemDB()
+	app := NewSimApp(log.NewTMLogger(log.NewSyncWriter(os.Stdout)), db, nil, true, map[int64]bool{}, 0)
+
+	genesisState := NewDefaultGenesisState()
+	stateBytes, err := codec.MarshalJSONIndent(app.Codec(), genesisState)
+	require.NoError(t, err)
+
+	// Initialize the chain
+	app.InitChain(
+		abci.RequestInitChain{
+			Validators:    []abci.ValidatorUpdate{},
+			AppStateBytes: stateBytes,
+		},
+	)
+	app.Commit()
+
+	// Making a new app object with the db, so that initchain hasn't been called
+	app2 := NewSimApp(log.NewTMLogger(log.NewSyncWriter(os.Stdout)), db, nil, true, map[int64]bool{}, 0)
+	//_, _, err = app2.ExportAppStateAndValidators(false, []string{})
+	//require.NoError(t, err, "ExportAppStateAndValidators should not have an error")
+	require.NotNil(t, app2)
+}
+
+// ensure that black listed addresses are properly set in bank keeper
+/* func TestBlackListedAddrs(t *testing.T) {
```

Comment:
What this all about?

Reply:
maybe just some clean up?

Reply:
I took the cosmos-sdk version of simapp stuff, and trimmed it down and modified it to match our app.  Some of this I could just remove, but need to talk about whether it is stuff we will support long-term or not.  I know some of it is functionality we just don't support yet.  Like Slashing and Genesis.  Others may not be relevant.  Probably need to have a discussion on some of the details.